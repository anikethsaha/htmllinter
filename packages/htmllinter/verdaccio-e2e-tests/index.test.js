const path = require('path');
const { spawnRegistry } = require('./helper/registry');
const { runNmp } = require('./helper/npm');
const { runCli } = require('./helper/cli');
const { getTempFolder } = require('./helper/fs-utils');
const pkg = require('../package.json');

jest.setTimeout(40000);

const version = pkg.version;

describe('E2E htmllinter cli', () => {
  let childFork;
  const folder = getTempFolder();
  beforeAll(async () => {
    childFork = await spawnRegistry();
    await runNmp(['publish', '--registry', 'http://localhost:4873']);

    await runNmp([
      'install',
      `@htmllinter/core@${version}`,
      '--prefix',
      folder,
      '--registry',
      'http://localhost:4873',
    ]);
  });

  afterAll(() => {
    if (childFork) {
      childFork.kill();
    }
  });

  it('should show the help output', async (done) => {
    const helpOutput = await runCli(
      path.join(folder, 'node_modules', '.bin', 'htmllinter'),
      []
    );
    expect(helpOutput).toMatchSnapshot();
    done();
  });

  it('should show the path of the input when input is passed but no linting rules', async (done) => {
    const output = await runCli(
      path.join(folder, 'node_modules', '.bin', 'htmllinter'),
      [path.join(__dirname, './fixture/simple/input.html')]
    );

    expect(output.split('/').slice(2, -1)).toMatchObject([
      'htmllinter',
      'packages',
      'htmllinter',
      'verdaccio-e2e-tests',
      'fixture',
      'simple',
    ]);
    done();
  });
});
