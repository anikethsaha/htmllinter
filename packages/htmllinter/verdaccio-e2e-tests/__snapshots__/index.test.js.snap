// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E htmllinter cli should show the help output 1`] = `
"
Usage :

$ htmllinter <input> [Options]

input           - it is the input(html) file name, it accepts glob pattern.
                example :
                  - \\"input.html\\"
                  - \\"input\\"
                  - \\"/*.html\\"
                  - \\".\\"

Options
  -c, --config    - path to htmllinter.config.js.
                    default is \\".\\"
                    example :
                    - '.'
                    - './dir/'

Example

    $ htmllinter input.html
    $ htmllinter input -c ../
    $ htmllinter **/*
"
`;
